design/stackhub.md

```diff
@@ -0,0 +1,207 @@
+
+## As-is
+
+- To introduce Kabanero topics, one must teach the student the concept of stacks, stack-hubs, pipelines, application development.  After all of that, then one must teach the student about collections and collection hubs, with a collection being a stack and a collection hub being a stack hub.  
+
+- To customize a stack, champ uses the appsody tools to copy the stack, and then has to manually convert it into a collection and include it in a collection hub.
+
+- To customize one character of a pipeline, I need to do a complete collection hub rebuild, which depending on how developer's are consuming collection hub releases may cause them to need to update all of their configuration -- for a change doesn't directly.
+
+## To-be
+-  Champ's (architect's) lifecycle changes to build stacks, using Appsody stackhub builds and independently building pipeline releases.
+-  Champ's experience is Kubernetes and operator-centric.  The configuration for Kabanero is defined solely by Kubernetes resources: `Kind:Kabanero` and `Kind:Stack`.
+-  Kabanero concepts are simplified to using stacks and their configuration.
+
+## Main Feature design
+
+### Custom Resource Changes
+
+#### Kind:Kabanero
+
+The Kabanero CRD will change to specifiy a Stack hub URL instead of a Collection Hub URL.  Previously, the Kabanero CRD did not concern itself with the relationship of pipelines with stacks.  The Kabanero CRD is enhanced to provide a default pipelines URL. 
+
+#### Kind:Stack
+
+The existing schema of the Collections CRD will be copied, with the `Kind:` field changing to `Stack`.  All other fields are relevant with one expection.  The status field `collections.url` field is being removed.  (Rationale below.)
+
+### Operator Managed Stacks vs. Manually Managed Stacks
+
+The Kabanero operator automatically manages stack configuration, pipeline configuration, etc. for those stacks that are discovered and configured in the Kabanero CR.   
+
+At times, it is desireable to manually instantiate a Stack CR, or override specific values (e.g. like pipelines) for a specific version of a specific stack-id.  An independent Stack CR can be applied to the configured Kabanero namespace, causing the specific pipelines to be installed and the Stacks to be activated.
+
+The stacks.URL field of the Kabanero CR is optional -- in this case, it indicates that the customer would like to manually manage all Stacks.
+
+The collections.url field in the Stacks CRD is no longer needed, and in the manual `stack` CR pattern above makes no sense, since the operator (or Git provided source) for the `stack` CR has provided all of the instance variables, i.e. the Stack was not derived from an `index.yaml` stack-hub.
+
```

Comment:
I think the same flexibility can be achieved through modifications to the Kabanero CRD structure thus making the need for this manual approach redundant. I really don't like the idea that my Kabanero CRD may not berepresentative of my currently active instance.

Reply:
We agreed that the approach that says Kabanero CR is the source of truth for configuration can be leveraged with this design for constrained environments like production -- and the design also supports some flexibility for the enterprise to change development environments w/o having to codify the changes in a single source of truth. 

Manually applied Stack CRs could be another approach for a single source of truth, where a single YAML file includes all of the Stack CR definitions and checked into Git -- applying the documents to the cluster making the configuration "so."